src/ophyd_async/core/_detector.py

```diff
@@ -267,6 +267,13 @@ async def read(self) -> dict[str, Reading]:
     async def describe(self) -> dict[str, DataKey]:
         return self._describe
 
+    @staticmethod
+    def ensure_trigger_info_exists(trigger_info: TriggerInfo | None) -> TriggerInfo:
```

Comment:
Please could this either be a private method which checks `self._trigger_info`:
```python
    def _ensure_trigger_info_exists(self) -> TriggerInfo:
```
Or a private free function that is passed `trigger_info`:
```python
def _ensure_trigger_info_exists(trigger_info: TriggerInfo | None) -> TriggerInfo:
```

Reply:
fixed 6860060329357f6adb3b4d3301bb84c8930de1e3